video/filter/vf_vapoursynth.c

```diff
@@ -208,6 +209,10 @@ static void copy_mp_to_vs_frame_props_map(struct priv *p, VSMap *map,
     if (img->fields & MP_IMGFIELD_INTERLACED)
         field = img->fields & MP_IMGFIELD_TOP_FIRST ? 2 : 1;
     p->vsapi->propSetInt(map, "_FieldBased", field, 0);
+
+    // Don't increase the reference count. It is not intended to be read externally,
+    // and we know it will be alive when we retrieve it.
+    p->vsapi->propSetData(map, "_MP_IMAGE", (const char *)img, sizeof(*img), 0);
```

Comment:
is this added so that VS scripts can mess with it somehow or what purpose does this serve?

Reply:
It is to get the input reference frame on output. VS script in practice should not access it as mp_image is internal to mpv and is not stable in terms of ABI.

If VS script wants to communicate changes in colorspace or read colospace info like DV, it has to be standardized in some other way, like #12593

Reply:
I see.
Wouldn't it be better here to store a pointer to `img` rather than a copy?

Also is the "it will be alive when we retrieve it" really guaranteed if VS scripts store and delay frames?

Reply:
Probably not. The issue is if we add reference and VS never gives us back this frame, it will leak memory of it. So either we do deep copy of everything in mp_image, including side_data or we just acknowledge that it has limitations of realtime processing and not some fancy timeshifting. We have pretty big buffered queue of 128 frames already, so it shouldn't be a problem for most "normal" scripts.

And sure probably